lib/Model/Importer/Binary/ELFImporter.cpp

```diff
@@ -429,17 +437,33 @@ void ELFImporter<T, HasAddend>::parseSymbols(object::ELFFile<T> &TheELF,
         or (Symbol.st_shndx == ELF::SHN_UNDEF))
       continue;
 
+    MetaAddress Address = MetaAddress::invalid();
     bool IsCode = Symbol.getType() == ELF::STT_FUNC;
-    if (!IsCode)
-      continue;
+    bool IsDataObject = Symbol.getType() == ELF::STT_OBJECT;
+    bool IsWeak = Symbol.getBinding() == ELF::STB_WEAK;
+    uint64_t Size = Symbol.st_size;
 
-    MetaAddress Address = MetaAddress::invalid();
-    Address = relocate(fromPC(Symbol.st_value));
-    auto It = Model->Functions.find(Address);
-    if (It == Model->Functions.end()) {
-      model::Function &Function = Model->Functions[Address];
-      if (MaybeName)
-        Function.OriginalName = *MaybeName;
+    if (IsCode)
+      Address = relocate(fromPC(Symbol.st_value));
+    else
+      Address = relocate(fromGeneric(Symbol.st_value));
+
+    if (IsCode) {
+      auto It = Model->Functions.find(Address);
+      if (It == Model->Functions.end()) {
+        model::Function &Function = Model->Functions[Address];
+        Function.Type = model::FunctionType::Regular;
+        if (MaybeName)
+          Function.OriginalName = *MaybeName;
+      }
+    }
+
+    if (MaybeName and IsDataObject and !IsWeak and Size > 0) {
```

Comment:
* [ ] Why do *need* a name?
* [ ] Why do you ignore weak symbols? I think we said not to do this.
* [ ] `s/!/not/`

Reply:
We may wish to ignore all symbols that overlap with previously introduced symbols (at the same address), but if a symbol is permitted to be overridden, we may want to use/show the strong one rather than the weak one (exactly like IDA and Ghidra do).

Reply:
Here you're not preferring the strong one if available, you're ignoring entirely the weak ones.

Make sure the other bullets in my comment are addressed too.

Reply:
> * [ ]  Why do _need_ a name?
> * [ ]  Why do you ignore weak symbols? I think we said not to do this.
> * [ ]  `s/!/not/`

1. We don't, removed.
2. Must have thought to handle only a specific set of cases that happened within revng tests. Removed `!IsWeak`, we cannot ignore weak symbols which may not have the strong one (linking phase might have not happened yet).
3. Sure.